lib/ueberauth/strategy/amco/plugs/authenticated_user.ex

```diff
@@ -30,10 +31,10 @@ defmodule Ueberauth.Strategy.Amco.Plugs.AuthenticatedUser do
     handler = Keyword.get(opts, :error_handler)
     source = Keyword.get(opts, :access_token_source)
 
-    with {:ok, token} <- get_access_token(conn, source),
-         {:ok, _response} <- validate_access_token(token) do
-      conn
-
+    with {:ok, access_token} <- get_access_token(conn, source),
+         {:ok, claims} <- API.userinfo(access_token) do
+      user = struct(User, claims)
```

Comment:
IMO we should whitelist what can the API can store in the User struct to avoid property injection. Imagine a client have a roles property defined in the User struct, they will be able to bump the logged user role up with any valid user.

Reply:
Agree. I improved it this, please take a look. Thanks.